### PR TITLE
Ignore commented out lines when finding features

### LINKED ISSUE #76246: Unstable Book: "foo" leads to an unrelated tracking issue, it's not a real feature
The Unstable Book in section [foo](https://doc.rust-lang.org/nightly/unstable-book/library-features/foo.html) leads to a tracking issue https://github.com/rust-lang/rust/issues/1234, which is about an unrelated topic `rust-mac1 bot has issues`. The naming of both the feature and the issue leads me to believe they are only given as examples in the code and are being incorrectly picked up by the documentation generator as real ones.

For instance they are used in `library/core/src/intrinsics.rs`:
```rust
//! In order to make an intrinsic usable at compile-time, one needs to copy the implementation
//! from https://github.com/rust-lang/miri/blob/master/src/shims/intrinsics.rs to
//! `librustc_mir/interpret/intrinsics.rs` and add a
//! `#[rustc_const_unstable(feature = "foo", issue = "01234")]` to the intrinsic.
```

### PR DESCRIPTION
**Pull request by syvb**
This fixes #76246, where commented out lines were being detected as features by `tidy`, by ignoring those lines when looking for features. It's still not perfect, since it can be fooled by things like:
```rust
/*
#[unstable(feature = "foo", issue = "1234")]
*/
```
But luckily that never happens in `rustc`, so `foo` now ceases to appear in the unstable book.

### CODE REVIEW

File: src/tools/tidy/src/features.rs [hunk 1]
```diff
@@ -423,6 +423,15 @@ fn map_lib_features(
                         continue;
                     }};
                 }
+
+                lazy_static::lazy_static! {
+                    static ref COMMENT_LINE: Regex = Regex::new(r"\s*//").unwrap();
```
Comment (jyn514) [inline src/tools/tidy/src/features.rs:428]: Shouldn't this have a starting anchor, so it avoids ignoring `let s = "//";`?
```suggestion
                    static ref COMMENT_LINE: Regex = Regex::new(r"^\s*//").unwrap();
```

Comment (jyn514) [reply src/tools/tidy/src/features.rs:428]: Also, there's a more accurate check in https://github.com/rust-lang/rust/blob/06f0adb34570ba83fee391abeb0bec0eec28a234/src/tools/tidy/src/style.rs#L81 - maybe these two checks should be combined?

Comment (Smittyvb) [reply src/tools/tidy/src/features.rs]: The other check is different enough (it looks for things *after* a comment) that it's probably better to leave them separate

**Comment by Mark-Simulacrum** [comment]
Seems OK. @bors r+ rollup

**Pull request merged**